src/sage/groups/perm_gps/permgroup.py

```diff
@@ -1581,6 +1581,74 @@ def smallest_moved_point(self):
         p = self._libgap_().SmallestMovedPoint()
         return self._domain_from_gap[Integer(p)]
 
+    @cached_method
+    def disjoint_direct_product_decomposition(self):
+        r"""
+        Returns the finest partition of `self.domain()` such that `self`
+        is isomorphic to the direct product of the projections of `self`
+        onto each part of the partition. Each part is a union of orbits
+        of `self`.
+
+        EXAMPLES::
+            sage: H = PermutationGroup([[(1,2,3),(7,9,8),(10,12,11)],[(4,5,6),(7,8,9),(10,11,12)],[(5,6),(8,9),(11,12)],[(7,8,9),(10,11,12)]])
+            sage: S = H.disjoint_direct_product_decomposition();S
+            {{1, 2, 3}, {4, 5, 6, 7, 8, 9, 10, 11, 12}}
+            sage: A = libgap.Stabilizer(H, list(S[0]), libgap.OnTuples);A
+            Group([ (7,8,9)(10,11,12), (5,6)(8,9)(11,12), (4,5,6)(7,8,9)(10,11,12) ])
+            sage: B = libgap.Stabilizer(H, list(S[1]), libgap.OnTuples);B
+            Group([ (1,2,3) ])
+            sage: T = PermutationGroup(gap_group=libgap.DirectProduct(A,B))
+            sage T.is_isomorphic(H)
+            True
+        """
+        from sage.combinat.set_partition import SetPartition
+        from sage.sets.disjoint_set import DisjointSet
+        H = self._libgap_()
+        if self.is_trivial():
+            return SetPartition(DisjointSet(self.domain()))
+        if libgap.NrMovedPoints(H) == self.degree() and libgap.IsTransitive(H):
+            return SetPartition([self.domain()])
+        O = libgap.Orbits(H)
+        k = len(O)
+        OrbitMapping = dict()
+        for i in range(k):
+            for x in O[i]:
+                OrbitMapping[x]=i
```

Comment:
before and after assignment `=` there should be a space (except when used for keyword arguments, and after `,` there should also be a space.